nx/lib/nx/defn/compiler.ex

```diff
@@ -124,6 +133,17 @@ defmodule Nx.Defn.Compiler do
     Process.get(Nx.Defn, false)
   end
 
+  @doc """
+  Returns a backend corresponding to the compiler options.
+
+  The backend matches the backend used for outputs from computations
+  defined by the given compiler.
+  """
+  def to_backend(opts) do
```

Comment:
I wasn't sure if compiler should be the first argument, but this matches the options we pass to `compile`/`jit`.

Reply:
`compile`/`jit` are private though. Should we expose this on `Nx.Defn` instead, which will also merge the given options with the global/local options, and then call the compiler?

Reply:
I meant `Nx.Defn.{compile,jit}`, yeah. Good call on merging! So `Nx.Defn.to_backend`?